Gemfile

```
@@ -19,7 +19,7 @@ gem 'rake', :group => [:development, :test]
 group :test do
   gem 'rspec'
   gem 'beaker', *location_for(ENV['BEAKER_VERSION'] || '~> 4.11')
-  gem "beaker-hostgenerator", *location_for(ENV['BEAKER_HOSTGENERATOR_VERSION'] || "~> 1.1")
+  gem "beaker-hostgenerator", *location_for(ENV['BEAKER_HOSTGENERATOR_VERSION'] || "~> 2.4.0")
```

We generally release BHG with new features as `2.y.0`, so I think you want the minor version to float like:

```suggestion
  gem "beaker-hostgenerator", *location_for(ENV['BEAKER_HOSTGENERATOR_VERSION'] || "~> 2.4")
```

---

Same page... I just did a similar review in pdb https://github.com/puppetlabs/puppetdb/pull/4000#discussion_r1759147238

---

Updated the code to float the minor version